### PR TITLE
Retrieve UID and GID from external experiment enpoint 

### LINKED ISSUE #37: Retrieve UID and GID from external experiment enpoint
Prepare code to retrieve uid and guid from external endpoint also having this from db-calipso.

### PR DESCRIPTION
**Pull request by acampsm**
First step to retrieve uid and gid from external endpoint and to be managed from db.

closes #37

@aicampbell you could add your code to get uid and gid from your service.

### CODE REVIEW

**Comment by aicampbell** [comment]
Duplicate of #33 
Please check the other pull request before submitting this.

**Comment by acampsm** [comment]
Now this PR has conflicts in CalipsoResourceDockerContainerService class

File: apprest/services/resource_docker.py [hunk 1]
```diff
@@ -53,16 +53,17 @@ def run_resource(self, username, experiment, public_name):
...
+            if not uid or not gid:
                 # Set the user's UID and GID
                 uid = user_service.get_user_uid(username)
                 gid = user_service.get_user_gid(username)
-            self.logger.debug('uid:%s, gid:%s' % (uid, gid))
+                self.logger.debug('uid or gid is None. Getting from username')
         except Exception as e:
-            self.logger.debug('uid,gid not found, set uid,gid to default')
+            self.logger.debug('Exception on get experiments,sessions, and uid,gid')
             uid = "1000"
```
Comment (aicampbell) [inline apprest/services/resource_docker.py]: The user method always sets the default UID and GID to 0 which is the same as running as root.
User 1000 doesn't always exist on images

**Comment by aicampbell** [comment]
If we aren't adding the UID and GID to the experiment model, can we remove it from the PR please?

File: apprest/services/resource_docker.py (hunk 1 above)
Comment (acampsm) [reply apprest/services/resource_docker.py]: These are default values if experiment not found.
If the experiment found but not values in uid or gid then they get from user_service.
If you have a better solution for this we could talk about it, we are flexible on this.

Comment (aicampbell) [reply apprest/services/resource_docker.py]: My suggestion is to set the UID and GID to 0.
The reason is that by default, every container can run as root.
However, if the user is not root, some containers will fail as some services require the user to be root

Comment (aicampbell) [reply apprest/services/resource_docker.py]: Plus by running as root, we are providing the exact same functionality as we did 2 weeks ago without adding UID and GID

Comment (acampsm) [reply apprest/services/resource_docker.py]: We have already donet

**Comment by acampsm** [comment]
Could you accept this PR? Now, does not have conflicts and we had been discussing about that.

**Pull request merged**